test-framework/sudo-test/src/lib.rs

```diff
@@ -150,9 +193,45 @@ impl EnvBuilder {
 
         let container = Container::new(BASE_IMAGE)?;
 
-        let mut groups = get_groups(&container)?;
         let mut users = get_users(&container)?;
 
+        for username in &users {
```

Comment:
this function has gotten quite gnarly. I'll look next into refactoring the sudo-test crate: introducing an intermediate representation should simplify this `build` function. I also want to tweak the API to be more "builder like" and avoid so many `None` and `&[]` arguments in method calls (in the compliance tests).